pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -345,8 +346,13 @@ def _process_training_step_output(self, training_step_output: STEP_OUTPUT) -> Op
         if isinstance(training_step_output, dict):
             loss = training_step_output.pop("loss", None)
             hiddens = training_step_output.pop("hiddens", None)
-            if hiddens is not None:
-                hiddens = hiddens.detach()
+
+            # handle cases of hiddens being a Tensor or a tuple of Tensors
+            def detach(tensor):
+                return tensor.detach()
+
+            hiddens = apply_to_collection(hiddens, dtype=Tensor, function=detach)
+
```

Comment:
This might not be necessary as we already detach all extras here

https://github.com/PyTorchLightning/pytorch-lightning/blob/c0caeb3ea9b1d889f32788f41f284ddaa2091c59/pytorch_lightning/trainer/connectors/logger_connector/result.py#L392-L407

I believe hiddens is considered part of the extras.

Can you check if that's right?

Reply:
I just tried to run with a LSTM using PL v 1.3.8 without my change but it does not seem to run:

`AttributeError: 'tuple' object has no attribute 'detach'`

Reply:
I did remove those lines and run an example with LSTM and indeed, it seems fine! I'll commit it.

Reply:
My bad, I believe it is necessary, we just did not test it.

Opened 8249 changing it back